### PR TITLE
Increasing land generation height

### PR DESCRIPTION
**Pull request by Dweblenod**
It will now match the 1.18 and beyond Overworld at y= -64 to 320. Changes have been made to the height at which the entry platform/island generates and to the gates, we will now be able to fit all 7 gates.

Values that have changed:
-16 blocks from top of entry platform to first gate  -> 4 blocks
-48 blocks between gates -> 30 blocks
-64 blocks between the highest gate and the top of the world -> 16 blocks
-top of entry platform generating at y=128 -> y=120

### CODE REVIEW

**Review by kirderf1 — CHANGES_REQUESTED** [review]
Changing hardcoded height values in worldgen features to new hardcoded values doesn't seem like a good solution. How about taking an approach with `Integer.MIN_VALUE` and `Integer.MAX_VALUE`?
On another regard, could you tweak the y gradient for the depth functions in `MSDensityFunctions`?

File: src/main/java/com/mraof/minestuck/world/gen/feature/FloorCogFeature.java [hunk 1]
```diff
@@ -30,7 +30,7 @@ protected ResourceLocation pickTemplate(Random random)
 	@Override
 	protected int pickY(WorldGenLevel level, BlockPos pos, Vec3i templateSize, Random random)
 	{
-		int yMin = Integer.MAX_VALUE, yMax = 0;
+		int yMin = Integer.MAX_VALUE, yMax = -64;
```
Comment (kirderf1) [inline src/main/java/com/mraof/minestuck/world/gen/feature/FloorCogFeature.java:33]:
```suggestion
		int yMin = level.getMaxBuildHeight(), yMax = level.getMinBuildHeight();
```

File: src/main/java/com/mraof/minestuck/world/gen/feature/OasisFeature.java [hunk 2]
```diff
@@ -61,7 +61,7 @@ public boolean place(FeaturePlaceContext<NoneFeatureConfiguration> context)
 		}
 		
 		//Figure out water level and if the ground is flat enough
-		int yMin = 256, yMax = 0;
+		int yMin = 320, yMax = -64;
```
Comment (kirderf1) [inline src/main/java/com/mraof/minestuck/world/gen/feature/OasisFeature.java:64]:
```suggestion
		int yMin = level.getMaxBuildHeight(), yMax = level.getMinBuildHeight();
```

File: src/main/java/com/mraof/minestuck/world/gen/feature/SmallLibraryFeature.java [hunk 3]
```diff
@@ -66,12 +66,12 @@ public boolean place(FeaturePlaceContext<NoneFeatureConfiguration> context)
 			door2 = new BoundingBox(z2, 0, minX, z2, 0, maxX);
 		}
 		
-		int y1 = 0;
+		int y1 = -64;
```
Comment (kirderf1) [inline src/main/java/com/mraof/minestuck/world/gen/feature/SmallLibraryFeature.java:69]:
```suggestion
		int y1 = level.getMinBuildHeight();
```

File: src/main/java/com/mraof/minestuck/world/gen/feature/SmallLibraryFeature.java [hunk 4]
```diff
@@ -66,12 +66,12 @@ public boolean place(FeaturePlaceContext<NoneFeatureConfiguration> context)
...
 		for(BlockPos doorPos : BlockPos.betweenClosed(door1.minX(), 0, door1.minZ(), door1.maxX(), 0, door1.maxZ()))
 		{
 			y1 = Math.max(y1, level.getHeight(Heightmap.Types.OCEAN_FLOOR_WG, pos.getX() + doorPos.getX(), pos.getZ() + doorPos.getZ()));
 		}
-		int y2 = 0;
+		int y2 = -64;
```
Comment (kirderf1) [inline src/main/java/com/mraof/minestuck/world/gen/feature/SmallLibraryFeature.java:74]:
```suggestion
		int y2 = level.getMinBuildHeight();
```

File: src/main/java/com/mraof/minestuck/world/gen/feature/SurfaceFossilsFeature.java [hunk 5]
```diff
@@ -39,7 +39,7 @@ protected ResourceLocation pickTemplate(Random random)
 	@Override
 	protected int pickY(WorldGenLevel level, BlockPos pos, Vec3i templateSize, Random random)
 	{
-		int yMin = Integer.MAX_VALUE, yMax = 0;
+		int yMin = Integer.MAX_VALUE, yMax = -64;
```
Comment (kirderf1) [inline src/main/java/com/mraof/minestuck/world/gen/feature/SurfaceFossilsFeature.java:42]:
```suggestion
		int yMin = level.getMaxBuildHeight(), yMax = level.getMinBuildHeight();
```

File: src/main/java/com/mraof/minestuck/world/gen/structure/SmallRuinPiece.java [hunk 6]
```diff
@@ -180,7 +180,7 @@ private boolean placeUnderling(int xPos, int zPos, BoundingBox boundingBox, Worl
 		if(!boundingBox.isInside(new BlockPos(xPos, 64, zPos)))
 			return false;
 		
-		int minY = 256, maxY = 0;
+		int minY = 320, maxY = -64;
```
Comment (kirderf1) [inline src/main/java/com/mraof/minestuck/world/gen/structure/SmallRuinPiece.java:183]:
```suggestion
		int minY = level.getMaxBuildHeight(), maxY = level.getMinBuildHeight();
```

**Review by kirderf1 — CHANGES_REQUESTED** [review]
You should probably also tweak placement range and frequencies of underground worldgen features, and potentially also carvers.

**Review by kirderf1 — APPROVED** [review]

**Comment by Dweblenod** [comment]
It has been mentioned in personal messages, but there is a delay to this because of bugs related to opening worlds in which chunks were loaded before the increase in world height

**Comment by Dweblenod** [comment]
When viewing all worlds, there is no error or warning message but upon attempting to load old worlds a warning message appears. We will not be fixing it so that old versions can be upgraded at the moment, but will try to make it absolutely clear that pre-1.18 worlds can not be opened without potential crashes or corruption

**Pull request merged**